src/TTimer.cpp

```diff
@@ -54,47 +61,42 @@ TTimer::TTimer(const QString& name, QTime time, Host* pHost)
 , mpTimer(new QTimer)
 , mModuleMember(false)
 {
-    mpTimer->stop();
+    mpQTimer->stop();
+    mpQTimer->setProperty(scmProperty_HostName, mpHost->getName());
+    mpHost->getTimerUnit()->mQTimerSet.insert(mpQTimer);
+    mpQTimer->setProperty(scmProperty_TTimerId, 0);
 }
 
 TTimer::~TTimer()
 {
-    mpTimer->stop();
-    if (!mpHost) {
-        return;
+    mpQTimer->stop();
+    if (mpHost) {
+        mpHost->getTimerUnit()->unregisterTimer(this);
     }
-    mpHost->getTimerUnit()->unregisterTimer(this);
-    mudlet::self()->unregisterTimer(mpTimer);
-    mpTimer->deleteLater();
-}
 
-bool TTimer::registerTimer()
-{
-    if (!mpHost) {
-        return false;
-    }
-    setTime(mTime);
-    mudlet::self()->registerTimer(this, mpTimer);
-    return mpHost->getTimerUnit()->registerTimer(this);
+    mpQTimer->deleteLater();
 }
 
 void TTimer::setName(const QString& name)
 {
     // temp timers do not need to check for names referring to multiple
-    // timer objects as names=ID -> much faster tempTimer creation
+    // timer objects as names=ID -> much faster tempQTimer creation
```

Comment:
Oops, a global find and replace shouldn't have touched this `mpTimer`...! :open_mouth:

Reply:
Fixed.